### PR TITLE
RCF-226

### PR DESCRIPTION
**Pull request by HimajaDhanyamraju2**
[RCF-226]

[RCF-226]: https://mosip.atlassian.net/browse/RCF-226?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pigeon.sh
Comment (jainhitesh9998) [inline pigeon.sh:1]: we need to update the command from flutter pub run to dart run
also we need to add the windows .bat for command line or .ps for powershell

File: lib/utils/app_config.dart [hunk 1]
```diff
@@ -49,6 +47,12 @@ FontWeight regular = FontWeight.w300;
 FontWeight bold = FontWeight.bold;
 FontWeight semiBold = FontWeight.w500;
 
+List<String> lang = [
```
Comment (jainhitesh9998) [inline lib/utils/app_config.dart:50]: we should add more description on what this means, it is a configuration for 1.1.5 compatibility, a description would help here

File: lib/ui/onboard/home_page.dart [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// ignore_for_file: use_build_context_synchronously, deprecated_member_use
+// ignore_for_file: deprecated_member_use
```
Comment (jainhitesh9998) [inline lib/ui/onboard/home_page.dart:1]: remove flutter lint ignore rules

File: lib/provider/global_provider.dart [hunk 3]
```diff
@@ -429,7 +405,8 @@ class GlobalProvider with ChangeNotifier {
   }
 
   List<LanguageData?> _languageDataList = [];
-  Map<String, String> _languageCodeMapper = {"eng": "English"};
+  Map<String, String> _codeToLanguageMapper = {"eng": "English"};
```
Comment (jainhitesh9998) [inline lib/provider/global_provider.dart:408]: this mapping we should get from a configuration file

File: android/app/src/main/java/io/mosip/registration_client/api_services/DynamicDetailsApi.java [hunk 4]
```diff
@@ -48,14 +51,14 @@ public void getFieldValues(@NonNull String fieldName, @NonNull String langCode,
...
-            List<GenericValueDto> genericValueList = this.masterDataService.findLocationByHierarchyLevel(hierarchyLevelName, langCode);
+            int level = Integer.parseInt(hierarchyLevelName);
+            List<GenericValueDto> genericValueList = this.masterDataService.findLocationByHierarchyLevel(level, langCode);
             genericValueList.forEach((v) -> {
                 DynamicResponsePigeon.GenericData location = new DynamicResponsePigeon.GenericData.Builder()
                         .setCode(v.getCode())
                         .setName(v.getName())
                         .setLangCode(v.getLangCode())
-                        .setHierarchyLevel((long) hierarchyLevel)
+                        .setHierarchyLevel((long) 0)
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/api_services/DynamicDetailsApi.java:61]: we should add explanation on why hierarchy level is changed to 0

File: android/app/src/main/java/io/mosip/registration_client/api_services/DynamicDetailsApi.java [hunk 5]
```diff
@@ -81,20 +84,20 @@ public void getDocumentValues(@NonNull String categoryCode, String applicantType
...
-            int hierarchyLevel = this.masterDataService.getHierarchyLevel(hierarchyLevelName);
-            List<GenericValueDto> genericValueList = this.masterDataService.findLocationByParentHierarchyCode(parentCode, langCode);
+            genericValueList = this.masterDataService.findLocationByParentHierarchyCode(parentCode, langCode);
             genericValueList.forEach((v) -> {
                 DynamicResponsePigeon.GenericData location = new DynamicResponsePigeon.GenericData.Builder()
                         .setCode(v.getCode())
                         .setName(v.getName())
                         .setLangCode(v.getLangCode())
-                        .setHierarchyLevel((long) hierarchyLevel)
+                        .setHierarchyLevel((long) 0)
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/api_services/DynamicDetailsApi.java:95]: we should add explanation on why hierarchy level is changed to 0

File: pigeon.sh
Comment (Piyush7034) [reply pigeon.sh:1]: Changed flutter pub to dart

File: lib/utils/app_config.dart (hunk 1 above)
Comment (Piyush7034) [reply lib/utils/app_config.dart:50]: Added description for language list.

File: lib/ui/onboard/home_page.dart (hunk 2 above)
Comment (Piyush7034) [reply lib/ui/onboard/home_page.dart:1]: Can't remove due to a dependency being used which is required.

File: lib/provider/global_provider.dart (hunk 3 above)
Comment (Piyush7034) [reply lib/provider/global_provider.dart:408]: Map getting populated with language data we are getting from database.

File: android/app/src/main/java/io/mosip/registration_client/api_services/DynamicDetailsApi.java (hunk 4 above)
Comment (Piyush7034) [reply android/app/src/main/java/io/mosip/registration_client/api_services/DynamicDetailsApi.java]: Removed from here as it is not being used now.

File: android/app/src/main/java/io/mosip/registration_client/api_services/DynamicDetailsApi.java (hunk 5 above)
Comment (Piyush7034) [reply android/app/src/main/java/io/mosip/registration_client/api_services/DynamicDetailsApi.java]: Removed from here as it is not being used now.

**Pull request merged**